app/src/main/java/com/sopt/now/utils/BaseResponse.kt

```diff
@@ -0,0 +1,12 @@
+package com.sopt.now.utils
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+abstract class BaseResponse {
+    @SerialName("code")
+    val code: String? = null
+    @SerialName("message")
+    val message: String? = null
+}
```

Comment:
[P3]
BaseResponse에 제네릭 타입의 data 변수를 추가해서,
다른 Response 클래스를 정의할 때 중복되는 코드(code, message)를 방지하는 것은 어떨까요?😊

```suggestion
@Serializable
data class BaseResponse<T> {
    @SerialName("code")
    val code: String? = null
    @SerialName("message")
    val message: String? = null
    @SerialName("data")
    val data: T? = null
}
```

Before
```
@Serializable
data class ResponseGetInfoDto(
    @SerialName("code")
    val code: Int,

    @SerialName("message")
    val message: String,

    @SerialName("data")
    val data: UserInfo
)

```

```
interface InfoService {
    @GET("member/info")
    suspend fun getUserInfo(
        @Header("memberId") memberId: String
    ): Response<ResponseGetInfoDto>
}
```

After

```
interface InfoService {
    @GET("member/info")
    suspend fun getUserInfo(
        @Header("memberId") memberId: String
    ): Response<BaseResponse<UserInfo>>
}
```

+ ResponseSignUpDto와 같이 code와 message로만 이루어진 Response는 BaseResponse로 대체할 수 있을 것 같습니다.😊